scanpy/preprocessing/_normalization.py

```diff
@@ -11,14 +11,16 @@
 
 def _normalize_data(X, counts, after=None, copy=False):
     X = X.copy() if copy else X
+    if issubclass(X.dtype.type, (int, np.integer)):
+        X = X.astype(np.float32)  # TODO: Check if float64 should be used
     after = np.median(counts[counts>0]) if after is None else after
     counts += (counts == 0)
-    counts /= after
+    counts = counts / after
     if issparse(X):
         sparsefuncs.inplace_row_scale(X, 1/counts)
     else:
-        X /= counts[:, None]
-    return X if copy else None
+        np.divide(X, counts[:, None], out=X)
```

Comment:
Why is this and the above change necessary? I though that if `__itruediv__` is not available, python falls back to `__truediv__` anyway?

Reply:
Sorry for the late response on this, totally forgot about it.

The conversion of `X` to float happens because you can't do an inplace dtype conversion. For example:

```
>>> a = np.ones(3, dtype=int)
>>> np.divide(a, 2, out=a)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: No loop matching the specified signature and casting was found for ufunc true_divide
```

This is also the reason behind changing `count /= after` with `counts = counts / after`. If `counts` was generated by `X.sum`, it could be an integer array. Which would do:

```
>>> a /= 2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: No loop matching the specified signature and casting was found for ufunc true_divide
```

Finally, I replaced `X /= counts[:, None]` with `np.divide(X, counts[:, None], out=X)` because if X was dask array `__itruediv__` doesn't behave like it does for numpy arrays https://github.com/dask/dask/issues/5199.

Reply:
Thank you! Would be cool to have this documented in a comment